File: output/ChatGPT/avatar/Java/Python/codeforces_459_A.py
def codeforces_459_A():
    x1, y1, x2, y2 = map(int, input().split())
    
    if x1 == x2:
        dif = abs(y1 - y2)
        print(x1 + dif, y1, x1 + dif, y2)
    
    elif y1 == y2:
        dif = abs(x1 - x2)
        print(x1, y1 + dif, x2, y2 + dif)
    
    elif abs(x1 - x2) == abs(y1 - y2):
        print(x1, y2, x2, y1)
    
    else:
        print(-1)

codeforces_459_A()

File: output/StarCoder/avatar/Python/Java/codeforces_459_A.java
import java.util.*;
public class codeforces_459_A {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String[] line = sc.nextLine().split(" ");
        int x1 = Integer.parseInt(line[0]);
        int y1 = Integer.parseInt(line[1]);
        int x2 = Integer.parseInt(line[2]);
        int y2 = Integer.parseInt(line[3]);
        int d = (int)Math.sqrt((x2-x1)*(x2-x1)+(y2-y1)*(y2-y1));
        if(x2-x1==0){
            System.out.println(x1+" "+(y1+d)+" "+x2+" "+(y2+d));
        }
        else if(y2-y1==0){
            System.out.println((x1+d)+" "+y1+" "+(x2+d)+" "+y2);
        }
        else if(x2-x1==y2-y1){
            System.out.println((x1+d)+" "+y1+" "+(x2+d)+" "+y2);
        }
        else if(x2-x1==-y2+y1){
            System.out.println((x1-d)+" "+y1+" "+(x2-d)+" "+y2);
        }
        else if(x2-x1==y2+y1){
            System.out.println((x1+d)+" "+(y1-d)+" "+(x2+d)+" "+(y2-d));
        }
        else if(x2-x1==-y2-y1){
            System.out.println((x1-d)+" "+(y1-d)+" "+(x2-d)+" "+(y2-d));
        }
        else{
            System.out.println("-1");
        }
    }
}

